From 7027240cb4825dd50337c818e448b24a7a60f8d1 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Thu, 01 Apr 2021 00:10:21 +0300
Subject: [PATCH] [SW-28697] version: enable read ver from modules

As a preparation for the version providing through the new LKD protocol
descriptor, enable reading the versions of Preboot (with BTL) and eFuse
internally.
Later on the LKD module should copy those to the descriptor.

Change-Id: I3c8dd536bde240ddfb7911e8e6e0f83a4a6c82cb
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/include/preboot.h b/include/preboot.h
index 5f2af5f..8b38daf 100644
--- a/include/preboot.h
+++ b/include/preboot.h
@@ -62,6 +62,11 @@
 	HL_DEBUG_HW_DBG_CARD,
 };
 
+enum preboot_vers {
+	PREBOOT_BTL_VER,
+	PREBOOT_EFUSE_VER,
+};
+
 extern uintptr_t _start;	/* Start of ppboot/preboot region */
 extern uintptr_t __stack_top;	/* End of ppboot/preboot region */
 
@@ -144,6 +149,7 @@
 void preboot_finalize_enabled_caps(void);
 void preboot_conf_iatu_access(bool enable);
 void preboot_disable_iatu_access(void);
+const char *preboot_get_version(enum preboot_vers ver);
 void hl_ram_init(void);
 bool hl_pci_link_is_up(void);
 
diff --git a/preboot/preboot.c b/preboot/preboot.c
index 48836fa..cf55488 100644
--- a/preboot/preboot.c
+++ b/preboot/preboot.c
@@ -91,11 +91,6 @@
 
 static const char *preboot_ver_str = "Preboot " VERSION_STR;
 
-enum preboot_vers {
-	PREBOOT_BTL_VER,
-	PREBOOT_EFUSE_VER,
-};
-
 #define BTL_VER_PREFIX	"BTL version "
 #define FUSE_VER_PREFIX	"Fuse version "
 
@@ -104,6 +99,11 @@
 	[PREBOOT_EFUSE_VER] = { FUSE_VER_PREFIX },
 };
 
+const char *preboot_get_version(enum preboot_vers ver)
+{
+	return preboot_vers[ver];
+}
+
 static void preboot_version(void)
 {
 	unsigned long offset = (unsigned long) preboot_vers[PREBOOT_BTL_VER];
